File: StarkAid.Api/Entities/DispositivoEsp.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace StarkAid.Api.Entities;

public class DispositivoEsp
{
    [Key] 
    public Guid Id { get; set; }

    [Required, MaxLength(150)] 
    public string Nome { get; set; } = string.Empty;

    [Required, MaxLength(45)] 
    public string Ip { get; set; } = string.Empty;

    [Required] 
    public int Porta { get; set; }

    [MaxLength(200)]
    public string? Comando { get; set; }

    [MaxLength(200)]
    public string? ComandToEsp { get; set; }

    [Required, MaxLength(50)]
    public string Status { get; set; } = "Desconectado";

    public bool LigadoDesligado { get; set; } = false;

    public Guid? UserId { get; set; }

    [Required] 
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset? LastPingAt { get; set; }

    public DateTimeOffset? LastUpdatedAt { get; set; }

    public User? User { get; set; }
}

File: starkaidautomacao/app/src/main/java/com/starkaid/starkaidapp/models/DispositivoEsp.kt
package com.starkaid.starkaidapp.models

import com.google.gson.annotations.SerializedName

data class DispositivoEsp(
    val id: String,
    val nome: String,
    val ip: String,
    val porta: Int,
    val comando: String?,
    @SerializedName("comandToEsp")
    val comandToEsp: String?,
    val status: String,
    val ligadoDesligado: Boolean,
    val userId: String? = null,
    val createdAt: String? = null,
    val lastPingAt: String? = null,
    val lastUpdatedAt: String? = null
)

data class CreateDispositivoEspRequest(
    val nome: String,
    val ip: String,
    val porta: Int,
    val comando: String? = null,
    @SerializedName("comandToEsp")
    val comandToEsp: String? = null
)

data class UpdateDispositivoEspRequest(
    val nome: String,
    val ip: String,
    val porta: Int,
    val comando: String? = null,
    @SerializedName("comandToEsp")
    val comandToEsp: String? = null,
    val status: String? = null,
    val ligadoDesligado: Boolean? = null
)

data class EnviarComandoRequest(
    val comando: String
)

data class EnviarComandoResponse(
    val dispositivo: DispositivoInfo,
    val comandoEnviado: String,
    val mensagem: String
)

data class DispositivoInfo(
    val nome: String,
    val ip: String,
    val porta: Int,
    val comando: String?,
    @SerializedName("comandToEsp")
    val comandToEsp: String?
)

data class PingResponse(
    val status: String,
    val isOnline: Boolean
)

